packages/synapse-react-client/src/components/AclEditor/AclEditor.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { Box, Collapse, Typography } from '@mui/material'
+import { ResourceAccessItem } from './ResourceAccessItem'
+import UserSearchBoxV2 from '../UserSearchBox/UserSearchBoxV2'
+import { ResourceAccess } from '@sage-bionetworks/synapse-types'
+import useUpdateAcl from './useUpdateAcl'
+import { TransitionGroup } from 'react-transition-group'
+import { PermissionLevel } from '../../utils/PermissionLevelToAccessType'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+export type AclEditorProps = {
+  isLoading: boolean
+  resourceAccessList: ResourceAccess[]
+  availablePermissionLevels: PermissionLevel[]
+  emptyText: React.ReactNode
+  addResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['addResourceAccessItem']
+  updateResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['updateResourceAccessItem']
+  removeResourceAccessItem: ReturnType<
+    typeof useUpdateAcl
+  >['removeResourceAccessItem']
+}
+
+export function AclEditor(props: AclEditorProps) {
+  const {
+    isLoading,
+    resourceAccessList,
+    availablePermissionLevels,
+    emptyText,
+    addResourceAccessItem,
+    updateResourceAccessItem,
+    removeResourceAccessItem,
+  } = props
+  if (isLoading) {
+    return (
+      <Box display={'flex'} justifyContent={'center'} height={'150px'}>
+        <SynapseSpinner size={50} />
+      </Box>
+    )
```

Comment:
Tweaked the loading UI to only show a spinner if loading. There was a skeleton in the old code but it was one rectangle & didn't block the rest of the UI, so this spinner is an improvement. May revisit with a more fleshed out skeleton later.